gammapy/modeling/models/core.py

```diff
@@ -996,6 +996,9 @@ def plot_regions(self, ax=None, kwargs_point=None, path_effect=None, **kwargs):
             WCS axes
         """
         regions = self.to_regions()
+        if isinstance(regions, list) and len(regions) == 0:
+            raise ValueError("Models don't have any associated region.")
```

Comment:
Should fail an error or simply log a warning that there are no regions to plot? I think the latter is more appropriate since having no region to plot is a still a regular usage.

Reply:
If we only raise a warning and then let the plotting function go on,it will fail with `ValueError("Region definition required.")` [here](https://github.com/gammapy/gammapy/blob/560589ab7c80a880401ccb444ecfa64bcbd6fad3/gammapy/maps/region/geom.py#L123). For this reason I raised this (more explicit) error before

Reply:
@adonath and @registerrier do you have further opinions about this? It is true that throwing an error here potentially breaks loops in which multiple `Models` regions are plotted, which could be annoying. But the alternative option (to plot a dummy region similarly to what happens when one calls `plt.plot()` with no data) would probably require the definition of some default WCS object

Reply:
I think the error is basically the right behavior. And also the error message of `ValueError: Region definition required.` is meaningful. Defining a default WCS seems not reasonable to me, it is completely arbitrary...

I would basically keep the current behavior but maybe allow the exception of `RegionGeom.plot()` to not fail if an `ax` is passed. Because in this case one can just show the image again without any region on top...